images-and-plotting-instructor.ipynb

```diff
@@ -0,0 +1,617 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Images and Image Plotting"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Arrays as images"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All photographic images represent a measurement of how much light hits the receiver. For instance, the Hubble image below is obtained by measuring the brightnesses of distant stars:\n",
+    "\n",
+    "![Hubble image](fig/galaxy.jpg)\n",
+    "\n",
+    "With traditional optical cameras, this measurement results in an image which is continuous, as it is projected directly onto paper. In order to store images digitally, they need to be divided into discrete chunks, pixels, each of which contains the value of the measurement in that small portion of the image. In this representation, an image is simply a grid of numbers, which allows it to be easily stored as an array with a shape equal to the resolution of the image.\n",
+    "\n",
+    "The `scikit-image` (abbreviated to `skimage` in code) module contains some sample images in the `data` submodule that we can use to demonstrate this principle."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Get some import statements out of the way.\n",
+    "from __future__ import division, print_function\n",
+    "%matplotlib inline\n",
+    "from matplotlib import pyplot as plt\n",
```

Comment:
we need to standardise on either:

```python
from matplotlib import pyplot as plt
```

or

```python
import matplotlib.pyplot as plt
```

personally I prefer the latter.